### ISSUE TITLE
False positive on `react-you-might-not-need-an-effect/no-pass-live-state-to-parent`

### ISSUE THREAD

**Issue by reteps** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#22]
We ran into a false positive passing refs to the parent component. Here is some related context:

```ts
export function SampleQuestionDemo({

  prompt,
  onMathjaxTypeset,
}: {
  prompt: ExamplePromptWithId;
  onMathjaxTypeset: (elements?: Element[]) => Promise<void>;

}) {
  useLayoutEffect(() => {
    if (cardRef.current) {
      // eslint-disable-next-line react-you-might-not-need-an-effect/no-pass-live-state-to-parent
      onMathjaxTypeset([cardRef.current]);Expand commentComment on line R86Code has comments. Press enter to view.
    }
  }, [variant?.question, onMathjaxTypeset])
```

https://github.com/PrairieLearn/PrairieLearn/pull/12378/files#r2195357088

> As far as I'm aware this is just a false positive. a) this isn't state per se, and b) the "state" isn't tracked by the parent in a way that would cause a rerender, which is what https://react.dev/learn/you-might-not-need-an-effect#notifying-parent-components-about-state-changes (linked to from the lint rule's readme) is concerned with.
> 
> This is possibly related to this TODO:
> 
> https://github.com/NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect/blob/5596e686debb751157418c1dacd5d2f8af1e70da/src/no-pass-live-state-to-parent.js#L46
> 
> Tbh I'm a little annoyed by this; I generally want lint rules to either give me correct and actionable advice or shut up and get out of the way. Given that you've already interacted with the maintainer of this plugin, consider reporting this upstream? I'm not entirely sure what to ask for here; maybe for refs to be ignored?
> 
> We could also solve this by structuring things such that we can just import `mathjaxTypeset` from this component and avoid the need to pass it in a prop. Regardless: I think the lint rule is wrong here.

**Comment by nwalters512** [comment]
🙏 cheers! Not at all disagreeing that this is a weird thing we're doing; it was self-inflicted by our own project structure and bundler configuration. It's definitely not idiomatic and we'll change this once we're able to, but I'm glad you're in agreement that whatever it is we're doing isn't really related to these rules.

**Comment by NickvanDyke** [comment]
Thanks for your patience and report!
